spec/features/users/opt_in_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe 'Email opt in', type: :feature do
+  let(:user)      { create(:user) }
+
+  context 'signed in' do
+    before { sign_in user }
+    after { sign_out user }
+
+    context 'in the user edit page' do
+      before { visit edit_user_path(user) }
+
+      shared_examples 'opting in' do |optin|
+        it 'allows opting in' do
```

Comment:
I may also be too paranoid here, but these tests look a bit too fine granular to me. They only check both extremes: no boxes are being checked and **a single one is checked**. For the latter scenario, the tests would also still be green if checking a single one would also automatically check all of the others 🤔 

So maybe a good trade of would just be a single long example instead of the shared one? I could for instance look like this:
```ruby
it 'allows opting in and out of email preferences' do
  # check that by default none are checked

  # check one or two and submit

  # navigate back to the page and check that the one (or two) are checked and all others aren't

  # opt out of one of the checked ones again

  # navigate back and check that the one you opted out is no longer checked
end
```